test/modules/losses/test_mdetr_losses.py

```diff
@@ -0,0 +1,118 @@
+
+    @pytest.fixture()
+    def total_boxes(self, n_boxes_per_sample):
+        return sum(n_boxes_per_sample)
+
+    @pytest.fixture()
+    def pred_logits(self, batch_size, num_queries, num_classes):
+        return torch.randn(batch_size, num_queries, num_classes + 1)
+
+    @pytest.fixture()
+    def max_slice_len(self):
+        return 3
+
+    @pytest.fixture()
+    def positive_map(self, max_slice_len, total_boxes, num_classes):
+        positive_map = torch.zeros(total_boxes, num_classes + 1)
+        for i in range(total_boxes):
+            start_idx = random.randint(0, num_classes - max_slice_len)
+            increment = random.randint(2, max_slice_len)
+            positive_map[i, start_idx : start_idx + increment] = 1
+        return positive_map
+
+    @pytest.fixture()
+    def indices(self, num_queries, n_boxes_per_sample):
+        return [
+            tuple(
+                torch.sort(
+                    torch.multinomial(
+                        torch.arange(num_queries, dtype=torch.float), n_boxes
+                    )
+                )
+            )
+            for n_boxes in n_boxes_per_sample
+        ]
+
+    @pytest.fixture()
+    def num_boxes(self, total_boxes):
+        return int(random.uniform(0.5 * total_boxes, 2 * total_boxes))
+
+    @pytest.fixture()
+    def construct_valid_boxes(self):
+        def _construct_valid_boxes(n_boxes):
+            boxes = []
+            for _ in range(n_boxes):
+                x1, y1 = torch.rand(2).unbind(-1)
+                x2 = random.uniform(x1.item(), 1)
+                y2 = random.uniform(y1.item(), 1)
+                box = box_convert(
+                    torch.Tensor([x1, y1, x2, y2]), in_fmt="xyxy", out_fmt="cxcywh"
+                )
+                boxes.append(box)
+            return torch.stack(boxes)
+
+        return _construct_valid_boxes
+
+    @pytest.fixture()
+    def pred_boxes(self, construct_valid_boxes, batch_size, num_queries):
+        return construct_valid_boxes(batch_size * num_queries).reshape(
+            batch_size, num_queries, -1
+        )
+
+    @pytest.fixture()
+    def target_boxes(self, construct_valid_boxes, n_boxes_per_sample):
+        return [construct_valid_boxes(n_boxes) for n_boxes in n_boxes_per_sample]
+
+    def test_soft_token_prediction_loss(
+        self, pred_logits, n_boxes_per_sample, positive_map, indices, num_boxes
+    ):
+        actual = torch.Tensor(
+            soft_token_prediction_loss(
+                pred_logits, n_boxes_per_sample, positive_map, indices, num_boxes
+            )
+        )
+        expected = torch.tensor(5.0893)
+        assert_expected(actual, expected, rtol=0, atol=1e-3)
+
+    def test_box_losses(self, pred_boxes, target_boxes, indices, num_boxes):
+        actual = box_losses(pred_boxes, target_boxes, indices, num_boxes)
+        expected_l1_loss = torch.tensor(0.7721)
+        expected_giou_loss = torch.tensor(1.1768)
```

Comment:
Hi @ebsmothers  May I know why fixed expected losses hard coded here. When adding new implementation for exponential (which will be called by multinomial and change the value of indices) , the fixed `expected_l1_loss = torch.tensor(0.7721)` and `expected_giou_loss = torch.tensor(1.1768)` is not applicable for new exponential.https://github.com/pytorch/pytorch/pull/69967

Reply:
@CaoE we have validated these fixed values against those from the original model, so by using these the goal is to ensure that the performance remains the same. Can you clarify what you mean by this? 

> the fixed expected_l1_loss = torch.tensor(0.7721) and expected_giou_loss = torch.tensor(1.1768) is not applicable for new exponential